packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/amplify-meta-updaters.ts

```diff
@@ -11,54 +11,54 @@ import { AuthParameters } from '../import/types';
-      !!triggers.CreateAuthChallenge &&
-      triggers.CreateAuthChallenge.length > 0 &&
-      !!triggers.VerifyAuthChallengeResponse &&
-      triggers.VerifyAuthChallengeResponse.length > 0;
-  }
-
-  options.customAuth = customAuthConfigured;
-  options.frontendAuthConfig = getFrontendConfig(authParameters);
+export const getPostAddAuthMetaUpdater =
+  (context: any, resultMetadata: { service: string; providerName: string }) =>
+  (resourceName: string): string => {
+    const options: any = {
+      service: resultMetadata.service,
+      providerPlugin: resultMetadata.providerName,
+    };
+    const parametersJSONPath = path.join(context.amplify.pathManager.getBackendDirPath(), 'auth', resourceName, 'parameters.json');
+    const authParameters = JSONUtilities.readJson<AuthParameters>(parametersJSONPath)!;
+
+    if (authParameters.dependsOn) {
+      options.dependsOn = authParameters.dependsOn;
+    }
 
-  context.amplify.updateamplifyMetaAfterResourceAdd('auth', resourceName, options);
+    let customAuthConfigured = false;
+    if (authParameters.triggers) {
+      const triggers = JSONUtilities.parse<any>(authParameters.triggers);
+
+      customAuthConfigured =
+        !!triggers.DefineAuthChallenge &&
+        triggers.DefineAuthChallenge.length > 0 &&
+        !!triggers.CreateAuthChallenge &&
+        triggers.CreateAuthChallenge.length > 0 &&
+        !!triggers.VerifyAuthChallengeResponse &&
+        triggers.VerifyAuthChallengeResponse.length > 0;
+    }
 
-  // Remove Identity Pool dependency attributes on userpool groups if Identity Pool not enabled
-  const allResources = context.amplify.getProjectMeta();
-  if (allResources.auth && allResources.auth.userPoolGroups) {
-    if (!authParameters.identityPoolName) {
-      const userPoolGroupDependsOn = [
-        {
-          category: 'auth',
-          resourceName,
-          attributes: ['UserPoolId', 'AppClientIDWeb', 'AppClientID'],
-        },
-      ];
-      context.amplify.updateamplifyMetaAfterResourceUpdate('auth', 'userPoolGroups', 'dependsOn', userPoolGroupDependsOn);
+    options.customAuth = customAuthConfigured;
+    options.frontendAuthConfig = getFrontendConfig(authParameters);
+
+    context.amplify.updateamplifyMetaAfterResourceAdd('auth', resourceName, options);
+
+    // Remove Identity Pool dependency attributes on userpool groups if Identity Pool not enabled
+    const allResources = context.amplify.getProjectMeta();
+    if (allResources.auth && allResources.auth.userPoolGroups) {
+      if (!authParameters.identityPoolName) {
+        const userPoolGroupDependsOn = [
+          {
+            category: 'auth',
+            resourceName,
+            attributes: ['UserPoolId', 'AppClientIDWeb', 'AppClientID'],
+          },
+        ];
+        context.amplify.updateamplifyMetaAfterResourceUpdate('auth', 'userPoolGroups', 'dependsOn', userPoolGroupDependsOn);
+      }
     }
-  }
-  return resourceName;
-};
+    return resourceName;
```

Comment:
GH shows difference here, but I cannot spot any, could you revert this section if there are no changes? Lines 22-60

Reply:
this changes seems to be triggered by a commit formatting hook, will update